Index: mini/method-to-ir.c
===================================================================
--- mini/method-to-ir.c	(revision 149264)
+++ mini/method-to-ir.c	(working copy)
@@ -56,6 +56,14 @@
 
 #include "ir-emit.h"
 
+#undef MONO_EMIT_BOUNDS_CHECK
+#define MONO_EMIT_BOUNDS_CHECK(cfg, array_reg, array_type, array_length_field, index_reg) do { \
+			(cfg)->flags |= MONO_CFG_HAS_ARRAY_ACCESS;					\
+			(cfg)->cbb->has_array_access = TRUE;						\
+    } while (0)
+
+#define MONO_ARCH_EMIT_BOUNDS_CHECK(cfg, array_reg, offset, index_reg) do { } while (0)
+
 #include "jit-icalls.h"
 #include "debugger-agent.h"
 
